test/libyul/Parser.cpp

```diff
@@ -304,6 +304,18 @@ BOOST_AUTO_TEST_CASE(if_statement_invalid)
 	BOOST_CHECK(successParse("{ if 42:u256 { } }"));
 }
 
+BOOST_AUTO_TEST_CASE(switch_case_types)
+{
+	CHECK_ERROR("{ switch 0:u256 case 0:u256 {} case 1:u32 {} }", TypeError, "Switch cases have non-matching types");
+	BOOST_CHECK(successParse("{ switch 0:u256 case 0:u32 {} case 1:u32 {} }"));
```

Comment:
Actually I think the type of the cases has to be equal to the type of the expression - but I think we cannot yet determine that. Could you please add a comment to the test?

Reply:
Yes, for that we'd need to determine types for expressions - I'll add a comment.